src/label-helpers.ts

```diff
@@ -22,37 +25,46 @@ function getTextContent(node) {
     .join('')
 }
 
-function getLabelContent(node) {
+function getLabelContent(node: Node | Element | HTMLInputElement) {
   let textContent
-  if (node.tagName.toLowerCase() === 'label') {
+  if ('tagName' in node && node.tagName.toLowerCase() === 'label') {
     textContent = getTextContent(node)
+  } else if ('value' in node) {
+    textContent = node.value
   } else {
-    textContent = node.value || node.textContent
+    textContent = node.textContent
   }
   return textContent
 }
 
 // Based on https://github.com/eps1lon/dom-accessibility-api/pull/352
-function getRealLabels(element) {
-  if (element.labels !== undefined) return element.labels ?? []
+function getRealLabels(element: Element | HTMLInputElement) {
+  // for old browsers
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+  if ('labels' in element && element.labels !== undefined)
+    return element.labels ?? []
 
   if (!isLabelable(element)) return []
 
   const labels = element.ownerDocument.querySelectorAll('label')
   return Array.from(labels).filter(label => label.control === element)
 }
 
-function isLabelable(element) {
+function isLabelable(element: Element) {
+  const labelableRegex = /BUTTON|METER|OUTPUT|PROGRESS|SELECT|TEXTAREA/
   return (
-    element.tagName.match(/BUTTON|METER|OUTPUT|PROGRESS|SELECT|TEXTAREA/) ||
+    labelableRegex.test(element.tagName) ||
     (element.tagName === 'INPUT' && element.getAttribute('type') !== 'hidden')
   )
 }
 
-function getLabels(container, element, {selector = '*'} = {}) {
-  const labelsId = element.getAttribute('aria-labelledby')
-    ? element.getAttribute('aria-labelledby').split(' ')
-    : []
+function getLabels(
+  container: Element,
+  element: Element,
+  {selector = '*'} = {},
+) {
+  const ariaLabelledBy = element.getAttribute('aria-labelledby')
+  const labelsId = isNotNull(ariaLabelledBy) ? ariaLabelledBy.split(' ') : []
```

Comment:
That's a change in behavior which we probably don't cover. For `<div aria-labelledby="">` we now return `[{ content: '', formControl: null }]` because `"".split(' ') => [""]`